### PR TITLE
[feat] Support test retries in the JUnit report

### LINKED ISSUE #1963: Support retries when generating a JUnit report
I think that we should support retries, because otherwise the test report is deceiving. Retries are used in scheduled runs mostly (not in CI), so the report must be accurate also in these cases.

### PR DESCRIPTION
**Pull request by teojgo**
* Each retry(run) is passed as an individual testsuite to the junit report.

Fixes #1963

### CODE REVIEW

**Review by vkarak — APPROVED** [review]
Lgtm. Do you have an example of how it shows up on Gitlab and Jenkins?

File: reframe/frontend/runreport.py [hunk 1]
```diff
@@ -159,51 +159,52 @@ def junit_xml_report(json_report):
...
-
-                # XSD schema does not like the exponential format and since we
-                # do not want to impose a fixed width, we pass it to `Decimal`
-                # to format it automatically.
-                'time': str(decimal.Decimal(tid['time_total'])),
+                'errors': '0',
+                'failures': str(rfm_run['num_failures']),
+                'hostname': json_report['session_info']['hostname'],
+                'id': str(run_id),
+                'name': f'ReFrame run {run_id + 1}',
```
Comment (vkarak) [inline reframe/frontend/runreport.py:170]:
```suggestion
                'name': f'ReFrame run {run_id}',
```

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]

**Review by vkarak — CHANGES_REQUESTED** [review]
We need also to remove the sentence in the documentation stating that this option does not work with the retries.

**Comment by jgphpc** [comment]
> Lgtm. Do you have an example of how it shows up on Gitlab and Jenkins?

see https://git.cscs.ch/c4k/c4kcust/example-junit/-/pipelines/26043/test_report

**Comment by jgphpc** [comment]
> Have you ran with the `--retries=<number>` option?

see https://git.cscs.ch/c4k/c4kcust/example-junit/-/pipelines/26045/test_report

**Comment by vkarak** [comment]
Thanks @jgphpc. It seems that Gitlab combines the results of everything. Anyway, it's up to each program how interprets the results, at least we encode everything.

**Comment by teojgo** [comment]
> Thanks @jgphpc. It seems that Gitlab combines the results of everything. Anyway, it's up to each program how interprets the results, at least we encode everything.

I could append a `run{id}` to the name of each test to differentiate.

**Comment by vkarak** [comment]
@jenkins-cscs retry daint dom

**Comment by vkarak** [comment]
> I could append a run{id} to the name of each test to differentiate.

No, that would mess with the actual test naming.

**Review by vkarak — APPROVED** [review]

**Pull request merged**